quran/fields.py

```diff
@@ -0,0 +1,82 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+
+
+class SegmentsField(models.TextField):
+    description = "Segments"
+
+    def to_python(self, value):
+        if value is None:
+            return None
+        # if value is empty string return empty list
+        if value == '':
+            return []
+        # if value is list
+        if isinstance(value, list):
+            # if list is not empty
+            if value:
```

Comment:
Your functions work (or should work!) for empty lists too, so there is no reason for this check.